src/test/java/com/google/sps/UploadImageTest.java

```diff
@@ -0,0 +1,63 @@
+package com.google.sps;
+
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.servlets.UploadImageServlet;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+// Responsible for testing the UploadImage and ImageHandler Servlets to make sure
+// They create accurate links.
+@RunWith(JUnit4.class)
+public final class UploadImageTest {
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private static final String EXPECTED_JSON_OUTPUT = "http://localhost:8080/_ah/upload/";
+  private static UploadImageServlet uploadImageServlet;
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+    MockitoAnnotations.initMocks(this);
+    this.uploadImageServlet = this.newUploadImageServlet();
+  }
+
+  // Creating local blobstore service
+  public UploadImageServlet newUploadImageServlet() {
+    BlobstoreService localBlobstore = BlobstoreServiceFactory.getBlobstoreService();
+    UploadImageServlet uploadImageServlet = new UploadImageServlet();
+    return uploadImageServlet;
+  }
+
+  @Test
+  public void uploadsProperly() throws IOException {
```

Comment:
This should be named for the method it's testing and the thing it's testing, so "doGet_uploadsImage"